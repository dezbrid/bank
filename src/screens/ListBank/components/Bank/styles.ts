import {StyleSheet} from 'react-native';

const HEIGHT_IMAGE = 90;
const WIDTH_IMAGE = 55;
const SIZES = {
  REGULAR: 15,
  TITLE: 17,
};
const COLORS = {
  codGray: '#0D0D0D',
  white: '#FFFFFF',
  tundora: '#4A4A4A',
};

export default StyleSheet.create({
  container: {
    flexDirection: 'row',
    borderRadius: 5,
    shadowOffset: {width: 0, height: 1},
    shadowOpacity: 1,
    shadowRadius: 1,
    shadowColor: COLORS.codGray,
    backgroundColor: COLORS.white,
    paddingVertical: 15,
    paddingHorizontal: 20,
    elevation: 3,
  },
  image: {
    height: HEIGHT_IMAGE,
    width: WIDTH_IMAGE,
  },
  containerText: {
    marginLeft: 15,
    flex: 1,
  },
  bankName: {
    fontSize: SIZES.TITLE,
    fontWeight: 'bold',
    color: COLORS.tundora,
  },
  description: {
    fontSize: SIZES.REGULAR,
    fontWeight: '500',
    color: COLORS.tundora,
  },
  ages: {
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'flex-end',
  },
});
