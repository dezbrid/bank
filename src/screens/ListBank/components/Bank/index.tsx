import React from 'react';
import {Text, View, Image} from 'react-native';
import {BankObject} from '@interfaces/bank';
import styles from './styles';

function Bank(bank: BankObject) {
  const {url, bankName, description} = bank;
  return (
    <View style={styles.container}>
      <Image style={styles.image} source={{uri: url}} resizeMode="stretch" />
      <View style={styles.containerText}>
        <Text style={styles.bankName}>{bankName}</Text>
        <Text style={styles.description}>{description}</Text>
      </View>
    </View>
  );
}
export default Bank;
