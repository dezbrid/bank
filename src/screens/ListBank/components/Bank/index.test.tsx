import 'react-native';
import React from 'react';
import {LISTBANKS} from '@mocks/bankMock';

import Bank from './index';
import {render, RenderResult} from '@testing-library/react-native';
let tree: RenderResult;
describe('<Bank />', () => {
  beforeEach(() => {
    tree = render(<Bank {...LISTBANKS[0]} />);
  });
  it('<Bank /> snapshot', () => {
    tree.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should exist title', () => {
    expect(tree.getByText(LISTBANKS[0].bankName)).toBeDefined();
  });
});
