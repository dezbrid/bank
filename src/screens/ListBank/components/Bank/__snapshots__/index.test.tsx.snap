// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Bank /> <Bank /> snapshot 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#FFFFFF",
      "borderRadius": 5,
      "elevation": 3,
      "flexDirection": "row",
      "paddingHorizontal": 20,
      "paddingVertical": 15,
      "shadowColor": "#0D0D0D",
      "shadowOffset": Object {
        "height": 1,
        "width": 0,
      },
      "shadowOpacity": 1,
      "shadowRadius": 1,
    }
  }
>
  <Image
    resizeMode="stretch"
    source={
      Object {
        "uri": "https://firebasestorage.googleapis.com/v0/b/stagingpagatodo-286214.appspot.com/o/Challenge%2Flogo-pagatodo.jpeg?alt=media&token=38b6ac4d-85ac-4288-bada-88eb5a0dec20",
      }
    }
    style={
      Object {
        "height": 90,
        "width": 55,
      }
    }
  />
  <View
    style={
      Object {
        "flex": 1,
        "marginLeft": 15,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#4A4A4A",
          "fontSize": 17,
          "fontWeight": "bold",
        }
      }
    >
      Paga Todo
    </Text>
    <Text
      style={
        Object {
          "color": "#4A4A4A",
          "fontSize": 15,
          "fontWeight": "500",
        }
      }
    >
      Banco Paga Todo es Para Todos
    </Text>
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "flex-end",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#4A4A4A",
            "fontSize": 17,
            "fontWeight": "bold",
          }
        }
      >
        edad: 
      </Text>
      <Text
        style={
          Object {
            "color": "#4A4A4A",
            "fontSize": 15,
            "fontWeight": "500",
          }
        }
      >
        10
      </Text>
    </View>
  </View>
</View>
`;
